lib/core_ext/numpy.rb

```diff
@@ -59,6 +75,59 @@ def roll(shift, axis: 0)
     end
   end
 
+  # set the passed value to all cells in the specified region
+  def full(r1, c1, r2, c2, value)
+    pts!(r1, c1, r2, c2)
+    r1.upto(r2) { |r| c1.upto(c2) { |c| self[r][c] = value } }
+
+    self
+  end
+
+  # set the cells starting at the specified coordinates with the contents of the passed array
+  def paste(r, c, other)
+    if r < 0 || c < 0 || c + other.first.length >= first.length || r + other.length >= length
+      raise ArgumentError, 'must be in bounds'
+    end
+
+    other.each_with_index { |row, r2| row.each_index { |c2| self[r2 + r][c2 + c] = other[r2][c2] } }
+
+    self
+  end
+
+  # add the passed difference to all values in the specified region
+  def add(r1, c1, r2, c2, difference)
```

Comment:
I still don't like this name. add is too much like append or concat. How about increment or inc?

Reply:
Also maybe use `n = 1` instead of `difference`